pennylane/ops/qubit/matrix_ops.py

```diff
@@ -292,8 +292,8 @@ def _eigvals(cls, *params):
         return D
 
     @staticmethod
-    def decomposition(D, wires):
-        return [QubitUnitary(qml.math.diag(D), wires=wires)]
+    def compute_decomposition(D, wires):
```

Comment:
I am wondering if I should do this for the matrix PR as well, to make the args in the signatures explicit!

Reply:
And with operation `__init__` methods. Then users can pass variables as keyword arguments like `qml.RX(theta=1.2, wires=0)`.  And it's a good check to restrict the number of arguments we receive.

Reply:
This is being done in master by @Jaybsoni as we speak!

Reply:
I updated the matrix PR to match yours!